integration_cdf_main.py

```diff
@@ -39,3 +39,23 @@
 conts = cdf.to_contours()
 print(conts)
 
+
+#%%
+annotations = cdf.filter(feature="roundness", min_val=0.5, max_val=1.0)
+print(annotations)
+
+# %%
+# Filter with inplace=True: CDF object is updated internally. Returns CDF object to allow chaining.
+updated_cdf = cdf.filter(
```

Comment:
this seems to be an inefficient why dont we give the filter multiple features and multiple gating parameters so it only has to loop ones